nawab.py

```diff
@@ -52,26 +57,65 @@ def wrapper(func, args, res):
 
 
 def main():
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument("-r", "--retweet", help="Retweet all tweets automatically, doesn't spawn a telegram bot",
+                        action='store_true', required=False)
+    parser.add_argument("-b", "--blacklist",type=list, required=False, help="Blacklist the given username")
+    parser.add_argument("-p", "--path", type=str, required=False,
+                        help="Path where the log files be stored. Note to create directory in that path beforehand.")
+    parser.add_argument('-V', '--verbose',action="store_const", const=20)
+    parser.add_argument('-s', '--silent', action="store_const", const=50)
+    parser.add_argument('-h', '--help', action='help', default=argparse.SUPPRESS,
+                        help='Show this help message and exit.')
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s 1.0', help="Show program's version number and exit.")
+     ## verbose is log level of Info with integer value 20 and silent is log level of critical of 
+     ## integer value of 50. The values are provided to futher assign  the level of logging.
+     ## Since logging level hold integer values.
+    args = vars(parser.parse_args())
+    ## set manual loglevels
+    if args['verbose']:
+        level = args['verbose']
+    elif args['silent']:
+        level = args['silent']
+    else:
+        ## Default mode: log level is Warning and above.for further detail visit
+        ## https://www.loggly.com/ultimate-guide/python-logging-basics/
+        level = logging.WARNING  
+    
     res = []
     data = pd.read_csv('data.csv')
     default_dir = '/var/log/nawab/'
-
+    ## blacklisting the names to current data.csv files
+    if args['blacklist']:
+        blist = pd.DataFrame({'Blacklist': args['blacklist']})
+        concats = pd.concat([data,blist])
+        concats.to_csv('data.csv', index=False)
+        ## if new path is passed then set new default path
+    if args['path']:
+        default_dir = args['path']
+   
     u_id = pwd.getpwuid(os.getuid()).pw_name
 
-    if not os.path.exists(default_dir):
-        os.system(("sudo mkdir %s" % (default_dir)))
-
     ownership_command = "sudo chown %s: %s" % (u_id, default_dir)
     os.system(ownership_command)
 
-    nawab = Nawab(default_dir,data)
+    nawab = Nawab(default_dir,data,level)
     #Initiate twitter bot
-    thread = threading.Thread(target=wrapper, args=(
-        nawab.twitter_bot_run, (), res))
-    thread.start()
+    if args['retweet']:
+        thread = threading.Thread(target=wrapper, args=(
+            nawab.twitter_bot_run, (), res))
+        thread.start()
+    else:
+        thread = threading.Thread(target=wrapper, args=(
+            nawab.twitter_bot_run, (), res))
+        thread.start()
```

Comment:
@iammarco11 correct me if im wrong but shouldn't the auto_retweet be passed as argument in thread's '()' of  if condition, if it is true?

Reply:
yes my mistake